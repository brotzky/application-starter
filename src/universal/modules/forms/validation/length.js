export default function(length) {
  return function(value) {
    const valueString = value && value.toString();
    return valueString.length === length
      ? undefined
      : {
          defaultMessage: 'should be {length} characters',
          id: 'form.errors.length',
          values: { length: length },
        };
  }
}
