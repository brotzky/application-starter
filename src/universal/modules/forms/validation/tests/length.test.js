import length from '../length';

const threeCharacters = length(3);

describe('[Form Validation] length', () => {
  describe('length()', () => {
    it('should return a function that will validate the length', () => {
      expect(length(3)).toMatchSnapshot();
    });

    it('should return no errors if a valid length string is passed in', () => {
      expect(threeCharacters('123')).toMatchSnapshot();
    });

    it('should return an error object when an invalid length is passed in', () => {
      expect(threeCharacters('1234')).toMatchSnapshot();
    });
  });
});
