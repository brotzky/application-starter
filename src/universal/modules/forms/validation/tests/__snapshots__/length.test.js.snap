// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] length length() should return a function that will validate the length 1`] = `[Function]`;

exports[`[Form Validation] length length() should return an error object when an invalid length is passed in 1`] = `
Object {
  "defaultMessage": "should be {length} characters",
  "id": "form.errors.length",
  "values": Object {
    "length": 3,
  },
}
`;

exports[`[Form Validation] length length() should return no errors if a valid length string is passed in 1`] = `undefined`;
